tests/user_extractor_test.php

```diff
@@ -159,11 +159,39 @@ public function test_get_user_case_insensitive() {
         global $CFG;
         $this->resetAfterTest();
 
-        // Arrange data
+        // Arrange data.
         $field = $this->add_user_profile_field('vehicleplate', 'text', true);
-        $expecteduser = $this->getDataGenerator()->create_user();
+        $expecteduser = $this->getDataGenerator()->create_user(['idnumber' => 'NB256']);
         profile_save_data((object)['id' => $expecteduser->id, 'profile_field_vehicleplate' => 'HD4999']);
 
+        // First, test get_user using a core field.
```

Comment:
I'd suggest to split this into two tests: one for profile fields and one for core fields.

Reply:
I've split the test into two. I'll wait to see where the db type discussion lands.